lib/srv/discovery/fetchers/db/aws_rds_proxy_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package db
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/cloud"
+	cloudtest "github.com/gravitational/teleport/lib/cloud/test"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+func TestRDSDBProxyFetcher(t *testing.T) {
+	rdsProxyVpc1, rdsProxyDatabaseVpc1 := makeRDSProxy(t, "rds-proxy-1", "us-east-1", "vpc1")
+	rdsProxyVpc2, rdsProxyDatabaseVpc2 := makeRDSProxy(t, "rds-proxy-2", "us-east-1", "vpc2")
+	rdsProxyEndpointVpc1, rdsProxyEndpointDatabaseVpc1 := makeRDSProxyCustomEndpoint(t, rdsProxyVpc1, "endpoint-1", "us-east-1")
+	rdsProxyEndpointVpc2, rdsProxyEndpointDatabaseVpc2 := makeRDSProxyCustomEndpoint(t, rdsProxyVpc2, "endpoint-2", "us-east-1")
+
+	clients := &cloud.TestCloudClients{
+		RDS: &cloudtest.RDSMock{
+			DBProxies:         []*rds.DBProxy{rdsProxyVpc1, rdsProxyVpc2},
+			DBProxyEndpoints:  []*rds.DBProxyEndpoint{rdsProxyEndpointVpc1, rdsProxyEndpointVpc2},
+			DBProxyTargetPort: 9999,
+		},
+	}
+
+	tests := []struct {
+		name          string
+		inputLabels   map[string]string
+		wantDatabases types.Databases
+	}{
+		{
+			name:          "fetch all",
+			inputLabels:   wildcardLabels,
+			wantDatabases: types.Databases{rdsProxyDatabaseVpc1, rdsProxyDatabaseVpc2, rdsProxyEndpointDatabaseVpc1, rdsProxyEndpointDatabaseVpc2},
+		},
+		{
+			name:          "fetch vpc1",
+			inputLabels:   map[string]string{"vpc-id": "vpc1"},
+			wantDatabases: types.Databases{rdsProxyDatabaseVpc1, rdsProxyEndpointDatabaseVpc1},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
I noticed you've parallelized the subtests but not the top level tests in this package, is that by design? I think this way none of the setup code runs in parallel

Reply:
To be honest I didn't give it too much thought. The tests in this lib took about half a second with or without parallel. The setup codes are very quick (just making mock structs) but the test cases are pretty quick too. Anyway, added parrallel to all tests for best parrallelization.